### PR TITLE
New feature: Filter for label

### PR DESCRIPTION
**Pull request by JEricaM**
Hi, I've implemented a new feature that let the user to filter the cards by labels. This filter works in combination with the member filter.

I would like you to consider adding this feature to the original git to allow everyone to use it.

Have a good day :)

### CODE REVIEW

File: src/app/calendar/calendar-card/calendar-card.component.html [hunk 1]
```diff
@@ -31,9 +31,8 @@
 <div class="content" >
 
   <div class="left">
-    <span *ngIf="flag_scaduto && !card.dueComplete" id="title_completed"> Expired activity</span>
-    <span *ngIf="!card.dueComplete" class="list" title="{{list?.name}}"><b>List:</b> {{list?.name}}</span>
-    <a *ngIf="!card.dueComplete" href="{{board?.url}}" class="board" target="_blank" title="{{board?.name}}"><b>Board:</b> {{board?.name}}</a>
+    <span *ngIf="!card.dueComplete" class="list" title="{{list?.name}}">{{list?.name}}</span>
+    <a *ngIf="!card.dueComplete" href="{{board?.url}}" class="board" target="_blank" title="{{board?.name}}">{{board?.name}}</a>
```
[the remark this answers is not in the document]
Comment (JEricaM) [reply src/app/calendar/calendar-card/calendar-card.component.html]: You can discard these, you've already deleted the list/board labels from the code.

**Comment by JEricaM** [comment]
Thank you :D Happy to help!

**Comment by JEricaM** [comment]
Why also this feature was deleted?

**Comment by CanKattwinkel** [comment]
@JEricaM 

It has caused performance problems. You have to imagine that some of our users are Trello Power users and therefore have hundreds to thousands of cards.

Of course, the feature is still desired, I just have to consider an alternative implementation.

**Comment by CanKattwinkel** [comment]
@JEricaM
 
I'm currently investigating this issue. To me it looks like that if we utilize the network to grab the labels we fire to many HTTP requests. The Trello API has a rate limit that is reached really fast. 

But if I look closer to the Cards Model there is `labels: { id: string, idBoard: string, name: string, color: string }[],` and `public idLabels?: string[],`. In fact i just fixed the type information since it was outdated. 

So we should be able to create a selector that reduces all labels of the cards that are currently in the view. It therefore consumes the `selectVisibleCardsInRange` selector and then reduces over the returned cards returning the label map.

**Comment by CanKattwinkel** [comment]
The changes were slightly modified, manually transferred to the master branch

**Pull request closed without merging**